app/src/main/java/org/simple/clinic/deeplink/DeepLinkEffectHandler.kt

```diff
@@ -0,0 +1,29 @@
+package org.simple.clinic.deeplink
+
+import com.spotify.mobius.rx2.RxMobius
+import io.reactivex.ObservableTransformer
+import org.simple.clinic.user.User
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import javax.inject.Inject
+
+class DeepLinkEffectHandler @Inject constructor(
+    private val userDao: User.RoomDao,
+    private val schedulerProvider: SchedulersProvider
+) {
+
+  fun build(): ObservableTransformer<DeepLinkEffect, DeepLinkEvent> = RxMobius
+      .subtypeEffectHandler<DeepLinkEffect, DeepLinkEvent>()
+      .addTransformer(FetchUser::class.java, fetchUser())
+      .build()
+
+  private fun fetchUser(): ObservableTransformer<FetchUser, DeepLinkEvent> {
+    return ObservableTransformer { effectsStream ->
+      effectsStream
+          .observeOn(schedulerProvider.io())
+          .map {
+            val user = userDao.userImmediate()
```

Comment:
Yes, I also check if the user object is present or not in the DB so that I can navigate them to setup activity. If the user object is present I then check for logged-in user state, the `userSession.isUserLoggedIn()` seems to be force casting the `Just` type so I wasn't sure if we should use that for this scenario where the user might not be logged in at all before clicking the deep link. WDYT?